### ISSUE TITLE
JavadocMethod accessModifiers incorrectly interpreted for interface class

### ISSUE THREAD

**Issue by ladislavlencucha** [checkstyle/checkstyle#10835]
I have read check documentation: https://checkstyle.org/config_javadoc.html#JavadocMethod
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac YOUR_FILE.java
#[[MAKE SURE THERE IS SUCCESSFUL COMPILATION]]
public interface MyInterface {
  /** Javadoc ok here. */
  void test();

  class MyClass {
    /** Missing parameter here. */
    public MyClass(Integer amount) {
    }
  }
}

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
"-//Puppy Crawl//DTD Check Configuration 1.2//EN"
"http://www.puppycrawl.com/dtds/configuration_1_2.dtd">
<module name="Checker">
  <property name="charset" value="UTF-8"/>
  <property name="fileExtensions" value="java, xml, properties"/>
  <module name="TreeWalker">
    <module name="JavadocMethod">
      <property name="accessModifiers" value="public"/>
      <property name="allowMissingParamTags" value="false"/>
      <property name="allowMissingReturnTag" value="true"/>
    </module>  
  </module>
</module>

/var/tmp $ cat YOUR_FILE.java
#[[PLACE YOU OUTPUT HERE]]

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.0-all.jar -c config.xml YOUR_FILE.java
Starting audit...
Audit done.
````
I expect checkstyle to report constructor to have missing javadoc param. Reason for that is that interface of this inner class is implicitly considered public according to java specs (Some IDEs even mark explicit public keyword as unnecessary).
If I add public keyword to class declaration, constructor is reported by checkstyle as expected.

**Comment by pbludov** [comment]
Similar to Issue #9876
The problem is [here](https://github.com/checkstyle/checkstyle/blob/5f1f940f78f4c208055657a2061b4a358c8ccae2/src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java#L469-L470):
```java
        // default access modifier
        AccessModifierOption accessModifier = AccessModifierOption.PACKAGE;
```

The default access modifier is not `package` in Java. The surrounding scope should be checked. Something [like that](https://github.com/checkstyle/checkstyle/blob/2877dbba42386e65396957b789f76e46a0030d99/src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java#L100-L120).

**Comment by Vyom-Yadav** [comment]
on it

**Comment by romani** [comment]
fix is merged
